### PR TITLE
Add unique IDs to plugins

### PR DESCRIPTION
**Pull request by ChristyKoh**
Make plugins identifiable by unique id so that multiple instances of the same plugin can be started, stopped, and have config saved correctly.

### CODE REVIEW

File: assistant/assistant.go [hunk 1]
```diff
@@ -219,19 +220,8 @@ func (assistant *Assistant) getPlugin(req *request) response {
...
-	case "AGGREGATOR":
-		data, err = assistant.agent.GetAggregatorPlugin(req.Plugin.Name)
-	case "PROCESSOR":
-		data, err = assistant.agent.GetProcessorPlugin(req.Plugin.Name)
-	default:
-		err = fmt.Errorf("did not provide a valid plugin type")
-	}
-
+	data, err = assistant.agent.GetRunningPlugin(req.Plugin.UniqueId)
+	fmt.Printf("Data: %+v", data)
```
Comment (TheGoldenKyle) [inline assistant/assistant.go]: Probably remove random debug prints, or make them ```log.Debug``` prints

File: assistant/assistant.go [hunk 2]
```diff
@@ -357,21 +348,26 @@ func (assistant *Assistant) stopPlugin(req *request) response {
...
 }
 
 // getRunningPlugins returns a JSON response obj with all running plugins
 func (assistant *Assistant) getRunningPlugins(req *request) response {
 	inputs := assistant.agent.GetRunningInputPlugins()
 	outputs := assistant.agent.GetRunningOutputPlugins()
-	data := pluginsList{inputs, outputs}
+	data := pluginsWithIdList{inputs, outputs}
 
 	res := response{SUCCESS, req.UUID, data}
```
Comment (TheGoldenKyle) [inline assistant/assistant.go]: Probably just return this instead of defining it and then returning.

**Review by TheGoldenKyle — APPROVED** [review]

File: agent/agent.go [hunk 3]
```diff
@@ -500,96 +477,99 @@ func (a *Agent) Run(ctx context.Context) error {
...
+
+	err = a.RunSingleInput(ri, a.Context)
+	if err != nil {
+		return "", err
+	}
+
+	// add new input to inputunit
+	a.iu.inputs = append(a.iu.inputs, ri)
+
+	a.Config.UpdateConfig(map[string]interface{}{"unique_id": uniqueId.String(), "name": pluginName}, uniqueId.String(), "inputs", "START_PLUGIN")
```
Comment (jodielu) [inline agent/agent.go]: Looks like UpdateConfig returns an error, need to add error handling

File: config/config.go [hunk 4]
```diff
@@ -729,6 +732,11 @@ func (c *Config) LoadConfig(path string) error {
 
 	c.serializeConfig(data, updatedConfigPath, map[string]interface{}{}, "", "", "")
```
Comment (jodielu) [inline config/config.go]: Note: Error handling for seralizeConfig

**Review by jodielu — APPROVED** [review]

**Review by TheGoldenKyle — APPROVED** [review]

**Pull request merged**